git/util.py

```diff
@@ -174,11 +174,16 @@ class RemoteProgress(object):
     DONE_TOKEN = 'done.'
     TOKEN_SEPARATOR = ', '
 
-    __slots__ = ("_cur_line", "_seen_ops")
+    __slots__ = ("_cur_line", "_seen_ops", "__progress_function")
     re_op_absolute = re.compile(r"(remote: )?([\w\s]+):\s+()(\d+)()(.*)")
     re_op_relative = re.compile(r"(remote: )?([\w\s]+):\s+(\d+)% \((\d+)/(\d+)\)(.*)")
 
-    def __init__(self):
+    def __init__(self, progress_function=None):
+        if progress_function is not None:
```

Comment:
Instead of the following 4 lines, does this work ?
```python
self.__progress_function = progress_function if progress_function else self.update
```